### ISSUE TITLE
Did something stupid, deleted a field and need to get the data back

### ISSUE THREAD

**Issue by ScAtMaN-UK** [mgibbs189/custom-field-suite#402]
I realise this is probably the wrong place to post this but need some advice.
(apologies for cross-posting from the wordpress.org forums)

I was playing around with adding and moving extra fields when I accidentally deleted one I need in the Edit field group screen. It was a textarea box added as part of a few other fields on ai1ec_event (time.ly calendar) page admin screens and would have contained data on many different event pages.
I tried to re-create it from the CFS edit fields screen but cant remember what the old field name was.

Will the data from the deleted field still be in the database? If so how can I find it’s field value and get it back and viewable in the edit post screens?

**Comment by chrisvanpatten** [comment]
Take a look at your custom meta fields. The key for the field should match the field name.

**Comment by ScAtMaN-UK** [comment]
Thanks for the reply Chris
I turned on Custom Fields under Screen Option on one of the event posts but could not see a custom field that would relate listed there. Am I looking on the right place?

**Comment by chrisvanpatten** [comment]
@mgibbs189 I just came across this on a client site. Is it expected behavior that deleting a field would also delete the meta? I thought it was the case that the postmeta would persist. Looking at a non-loop field in particular.

**Comment by mgibbs189** [comment]
@chrisvanpatten Yes, it's expected behavior: https://github.com/mgibbs189/custom-field-suite/blob/master/includes/field_group.php#L254-L264

**Comment by chrisvanpatten** [comment]
Now fixed by #443 for future reference! To prevent CFS from deleting post meta when deleting fields, you can use a snippet like this:

```
add_filter( 'cfs_deleted_field_ids', function ($deleted_field_ids) {
    return [];
} );
```
